### DISCUSSION TITLE
Create a trivy plugin for own CVE query

### DISCUSSION THREAD

**Discussion by alfeilex** [aquasecurity/trivy#7740] [Q&A]
### Question

Hello, 

I would like to ask if it's possible to create a trivy plugin that uses the query components to search for specific CVE's that match a vendor, or product and version. Is there an API to using the database for other languages like Java? 

Finally, we want to include trivy in our CLI tool, but we are looking for a functionality so that the user can search for a CVE given the vendor/product and version. The output for that query should include matching CVE's and perhaps references/fixed versions and severity.

Thank you 

### Target

None

### Scanner

Vulnerability

### Output Format

JSON

### Mode

None

**Comment by itaysk** [comment]
this is out of scope for trivy. you can try to query the trivy db directly

**Comment by knqyf263** [comment]
I'd recommend [boltwiz](https://github.com/Moniseeta/boltwiz) to query Trivy DB.
